src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_StringInterpolation.cs

```diff
@@ -218,6 +249,24 @@ private void MakeInterpolatedStringFormat(BoundInterpolatedString node, out Boun
                 var part = node.Parts[i];
                 if (part is BoundStringInsert fillin)
                 {
+                    // string or char constants without alignment or format
+                    if (canHideOptimization && fillin is { Alignment: null, Format: null, Value.ConstantValueOpt: { } constantValueOpt })
+                    {
+                        switch (constantValueOpt)
+                        {
+                            case { IsChar: true, CharValue: char ch }:
+                                stringBuilder.Append(escapeInterpolatedStringLiteral(ch.ToString()));
+                                continue;
+
+                            case { IsString: true, StringValue: var str }:
+                                stringBuilder.Append(escapeInterpolatedStringLiteral(str ?? ""));
+                                continue;
+
+                            case { IsNull: true }:
+                                // null literal for string? type
+                                continue;
+                        }
```

Comment:
We can't do this generally. We can do it for interpolated strings that are directly observed as `string`, but for `IFormattable` or `FormattableString`, this will change the observable semantics of the returned formattable string (as you can call `GetArguments()` and do post-processing on them if you want). If this is what you meant by the `canHideOptimizations` parameter, then I would suggest calling that parameter something like `usingDirectlyAsString` instead. Also, please leave a comment with this explanation in the code.

Reply:
@333fred 
Should we consider `FormattableString.Invariant`?

> `usingDirectlyAsString`

You seem to have thought "using" is better than "used". If so I will follow you because there do not seem to be a better candidate that is not much longer.

Reply:
`observedAsString`?